File: clients/python/examples/workflow_auto_launch.py
from __future__ import annotations

from sikuligo import Pattern, Screen

# Workflow: client auto-launches sikuligo if needed.
screen = Screen.start()
try:
    match = screen.click(Pattern("assets/pattern.png").exact())
    print(f"clicked match target at ({match.target_x}, {match.target_y})")
finally:
    screen.close()

File: clients/node/examples/workflow-auto-launch.js
import { Screen, Pattern } from "../src";

async function main() {
  // Workflow 1: client auto-launches sikuligo if needed.
  const screen = await Screen.start();
  try {
    const match = await screen.click(new Pattern("assets/pattern.png").exact());
    console.log(`clicked match target at (${match.targetX}, ${match.targetY})`);
  } finally {
    await screen.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
